### DISCUSSION TITLE
Multiple join query

### DISCUSSION THREAD

**Discussion by samsamai** [SeaQL/sea-orm#50] [Q&A] [answered]
@tyt2y3 @billy1624 I've got a bit stuck while trying to test a more complicated query in the test suite.
How can I do a query such as this with ORM:

```
SELECT
	baker.name, sum(lineitem.price * lineitem.quantity)

FROM
	baker
	INNER JOIN cakes_bakers ON cakes_bakers.baker_id = baker.id
	INNER JOIN cake ON cakes_bakers.cake_id = cake.id
	INNER JOIN lineitem ON lineitem.cake_id = cake.id
GROUP BY
	baker.name;
```

**Comment by tyt2y3** [comment, marked as answer]
https://github.com/SeaQL/sea-orm/blob/ca9aa01f6b1bc53e4b089bf8aa4c5775b1e54d8b/src/query/helper.rs#L130
The `join` method on the `QuerySelect` trait is not bounded by `Related` trait. `RelationDef` can be constructed at runtime too.

**Reply by tyt2y3** [reply]
Does that work for you?

**Reply by samsamai** [reply]
I got further in but not quite there so I decided to simplify the test as it was to only test `having`.

With `having` it would be useful to implement PartialOrd so we could do comparisons e.g SomeColumn.gt(some_value) unless there is a way to do that already which I couldn't find.

**Reply by tyt2y3** [reply]
is this what you are looking for?

https://github.com/SeaQL/sea-orm/blob/ca9aa01f6b1bc53e4b089bf8aa4c5775b1e54d8b/src/query/helper.rs#L91-L104

**Reply by samsamai** [reply]
Yes similar to that but instead of `equal` I wanted to do `greater than`.
Currently PartialEq is implemented for ActiveValue<V> which allows us to do `.having(cake::Column::Id.eq(4))` but PartialOrd is not implemented for ActiveValue<V> so we can't do `.having(cake::Column::price.gt(dec!(50.0)))`

**Reply by tyt2y3** [reply]
https://github.com/SeaQL/sea-orm/blob/80976116de6b93a27e2074f9a51157993dd7f29b/src/entity/column.rs#L83-L88

I am sorry I did not get the problem. I thought all operators eq, gt, lt, etc are implemented together. Is it a limitation of SeaQuery?

**Reply by samsamai** [reply]
Sorry my mistake, I just tried `.having(order::Column::Total.gt(dec!(100.00)))` in one of the tests and it works!
It must have failed for me before because the previous version of the test was trying to do multiple joins or something, anyway it works, so all good!

**Reply by tyt2y3** [reply]
Thanks, that's great!
